language/move-vm/runtime/src/native_functions.rs

```diff
@@ -173,3 +195,49 @@ impl<'a, 'b> NativeContext<'a, 'b> {
         self.interpreter.get_stack_frames(count)
     }
 }
+
+pub struct NativeContextDataStore<'a> {
+    data_store: &'a mut dyn DataStore,
+}
+
+impl<'a> DataStore for NativeContextDataStore<'a> {
```

Comment:
All the accesses here bypass VM caches/loader and versioning. Besides performance, I'm afraid this will not work with parallel execution. @gelash 

Can we perhaps discuss the use case on slack first?